docs/misc/str-builder-api.rst

```diff
@@ -0,0 +1,390 @@
+bytes/str building API
+=======================
+
+We need to design an HPy API to build ``bytes`` and ``str`` objects. Before making
+any proposal, it is useful to understand:
+
+1. What is the current API to build strings.
+
+2. What are the constraints for alternative implementations and the problems
+   of the current C API.
+
+3. What are the patterns used to build string objects in the existing
+   extensions.
+
+Some terminology:
+
+    * "string" means both ``bytes`` and ``str`` objects
+    * "unicode" or "unicode string" indicates ``str``
+
+.. note::
+   In this document we are translating ``PyUnicode_*`` functions into
+   ``HPyStr_*``. See issue #xxx for more discussion about the naming convention.
```

Comment:
My reading of this is that HPy would use `HPyBytes_*` and `HPyStr_*` - use the Python name for the type since we don't need to distinguish from the Py2 string type.

Reply:
yes exactly, my proposal will be to have `HPyBytes_*` and `HPyStr_*`, as opposed to `HPyBytes_*` and `HPyUnicode_*`.
I'll open an issue for that so we can discuss the naming in a separate place than the API itself